packages/jsii/lib/reserved-words.ts

```diff
@@ -184,3 +188,71 @@ export const PYTHON_RESERVED = new Set([
   'raise',
   'self',
 ]);
+
+export const GO_RESERVED = new Set([
```

Comment:
Can we remove the [`RESERVED_WORDS` map in golang/utils](https://github.com/aws/jsii/blob/master/packages/jsii-pacmak/lib/targets/golang/util.ts#L49) and change the logic inside of [`substituteReservedWords`](https://github.com/aws/jsii/blob/master/packages/jsii-pacmak/lib/targets/golang/util.ts#L55) to use this as the source?